src/iot/explorer/iotCertificateNode.ts

```diff
@@ -0,0 +1,191 @@
+    public constructor(
+        public readonly certificate: IotCertificate,
+        public readonly parent: IotCertsFolderNode | IotThingNode,
+        public readonly iot: IotClient,
+        collapsibleState: vscode.TreeItemCollapsibleState,
+        protected readonly workspace = Workspace.vscode()
+    ) {
+        //Show only 8 characters in the explorer instead of the full 64. The entire
+        //ID can be copied from the context menu or viewed when hovered over.
+        super(certificate.id.substring(0, 8), collapsibleState)
+        this.tooltip = localize(
+            'AWS.explorerNode.iot.fileTooltip',
+            '{0}\nStatus: {1}\nCreated: {2}',
+            this.certificate.id,
+            this.certificate.activeStatus,
+            moment(this.certificate.creationDate).format(S3_DATE_FORMAT)
+        )
+        this.description = `\t[${this.certificate.activeStatus}]`
+        this.contextValue = `${CONTEXT_BASE}.${this.certificate.activeStatus}`
+    }
+
+    /**
+     * See {@link IotClient.updateCertificate}
+     */
+    public async activate(): Promise<void> {
+        await this.iot.updateCertificate({ certificateId: this.certificate.id, newStatus: STATUS_ACTIVE })
+    }
+
+    /**
+     * See {@link IotClient.updateCertificate}
+     */
+    public async deactivate(): Promise<void> {
+        await this.iot.updateCertificate({ certificateId: this.certificate.id, newStatus: STATUS_INACTIVE })
+    }
+
+    /**
+     * See {@link IotClient.updateCertificate}
+     */
+    public async revoke(): Promise<void> {
+        await this.iot.updateCertificate({ certificateId: this.certificate.id, newStatus: STATUS_REVOKED })
+    }
+
+    /**
+     * See {@link IotClient.attachPolicy}
+     */
+    public async attachPolicy(policyName: string): Promise<void> {
+        await this.iot.attachPolicy({ policyName: policyName, target: this.certificate.arn })
+    }
+
+    public update(): void {
+        return undefined
+    }
+
+    public get arn(): string {
+        return this.certificate.arn
+    }
+
+    public get name(): string {
+        return this.certificate.id
+    }
+
+    public [inspect.custom](): string {
+        return `IotCertificateNode (certificate=${this.certificate.id})`
+    }
+}
+
+export class IotThingCertNode extends IotCertificateNode {
+    public constructor(
+        public readonly certificate: IotCertificate,
+        public readonly parent: IotThingNode,
+        public readonly iot: IotClient,
+        protected readonly workspace = Workspace.vscode()
+    ) {
+        super(certificate, parent, iot, vscode.TreeItemCollapsibleState.None, workspace)
+        this.contextValue = `${CONTEXT_BASE}.Things.${this.certificate.activeStatus}`
+    }
+
+    /**
+     * See {@link IotClient.detachThingPrincipal}
+     */
+    public async detachThing(): Promise<void> {
+        await this.iot.detachThingPrincipal({ thingName: this.parent.thing.name, principal: this.certificate.arn })
+    }
```

Comment:
General comment: I have a preference towards these nodes being limited to presentation and data abstraction. Functionality performing any other kind of business logic or actions on the remote resource should live with commands.

Reply:
Removed these from the explorer nodes